analytics.js

```diff
@@ -706,25 +765,61 @@ class Analytics {
           this.clientIntegrationObjects,
         );
 
-        // try to first send to all integrations, if list populated from BE
-        succesfulLoadedIntersectClientSuppliedIntegrations.forEach((obj) => {
-          try {
-            if (!obj.isFailed || !obj.isFailed()) {
-              if (obj[type]) {
-                const sendEvent = !this.IsEventBlackListed(rudderElement.message.event, obj.name);
-
-                // Block the event if it is blacklisted for the device-mode destination
-                if (sendEvent) {
-                  const clonedRudderElement = cloneDeep(rudderElement);
-                  obj[type](clonedRudderElement);
-                }
-              }
+        const intgWithoutTransformation = [];
+        const intgWithTransformation = [];
+
+        // Depending on transformation is connected or not
+        // create two sets of destinations
+        succesfulLoadedIntersectClientSuppliedIntegrations.forEach((intg) => {
+          const sendEvent = !this.IsEventBlackListed(rudderElement.message.event, intg.name);
+
+          // Block the event if it is blacklisted for the device-mode destination
+          if (sendEvent) {
+            if (intg.areTransformationsConnected) {
+              intgWithTransformation.push(intg);
+            } else {
+              intgWithoutTransformation.push(intg);
             }
-          } catch (err) {
-            err.message = `[sendToNative]::[Destination:${obj.name}]:: ${err}`;
-            handleError(err);
           }
         });
+
+        // loop through destinations that doesn't have
+        // transformation connected with it and send events
+        intgWithoutTransformation.forEach((intg) => {
+          this.sendDataToDestination(intg, rudderElement, type);
+        });
+
+        if (intgWithTransformation.length) {
+          try {
+            // Process Transformation
+            processTransformation(
```

Comment:
We need to change the implementation to use event queue.

Reply:
Addressed